File: back/api/Models/Review/ProductReview.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models.Review;

public class ProductReview
{
  [Key] public Guid Id { get; init; }
  [Required] public int ProductId { get; init; }
  [Required] public string ReviewersNickname { get; init; }
  [Required] public bool ByEmployee { get; init; }
  [Required] public string Title { get; init; }
  [Required] public string Content { get; init; }
  [Required] public int Stars { get; init; }
  [Required] public DateTimeOffset CreatedAt { get; init; }
  [Required] public bool IsApproved { get; set; }

  public virtual IEnumerable<ProductReviewComment> Comments { get; init; }
}

File: front/types/ProductReview.ts
export type ProductReview = {
  id: string;
  productId: number;
  reviewersNickname: string;
  byEmployee: boolean;
  title: string;
  content: string;
  stars: number;
  createdAt: string;
};

export type AddProductReviewRequestBody = Omit<
  ProductReview,
  "id" | "productId" | "byEmployee" | "createdAt"
>;

export type ProductReviewComment = {
  id: string;
  reviewId: string;
  commentersNickname: string;
  byEmployee: string;
  title: string;
  content: string;
  createdAt: string;
};

export type AddProductReviewCommentRequestBody = Omit<
  ProductReviewComment,
  "id" | "reviewId" | "byEmployee" | "createdAt"
>;
